lib/hash_ring/ets.ex

```diff
@@ -252,21 +288,47 @@ defmodule ExHashRing.HashRing.ETS do
       {:reply, {:error, :node_exists}, state}
     else
       nodes = [{node_name, num_replicas} | nodes]
-      {:reply, {:ok, nodes}, rebuild(%{state | nodes: nodes})}
+      {:reply, {:ok, nodes}, update_nodes(state, nodes)}
+    end
+  end
+
+  def handle_call({:add_nodes, nodes}, _from, %__MODULE__{} = state) do
+    nodes = transform_nodes(nodes, state.default_num_replicas)
+
+    existing_nodes = Enum.filter(nodes, fn {name, _} ->
```

Comment:
perf nit: seems like the flow of this function could be reordered into:
1. are there existing nodes that overlap with _any_ of the new nodes?
2. Yes? abort and return error
3. No? add nodes to state.nodes

Reply:
Changed this to use Enum.any?/2 in https://github.com/discord/ex_hash_ring/pull/10/commits/173fbda6fff6a11075af73ad466eed548991667b